common/workunit/workunit.cpp

```diff
@@ -2881,19 +2881,127 @@ class CDaliWorkUnitFactory : public CWorkUnitFactory, implements IDaliClientShut
                                                 ISecManager *secmgr,
                                                 ISecUser *secuser)
     {
+        class CQueryOrFilter : public CInterface, implements IInterface
```

Comment:
I don't see the benefit of this use of implementing IInterface - especially if it is not used as an array.  Just deriving from CInterface is cleaner (or possibly CInterfaceOf<IIntrerface>).